### ISSUE TITLE
[TypeSpec] Support @flattenProperty

### ISSUE THREAD

**Issue by lmazuel** [Azure/autorest.typescript#2164]
Generate flattened models when the TypeSpec contains this annotation

Needs TCGC: https://github.com/Azure/typespec-azure/issues/13
Needs cadl-ranch: https://github.com/Azure/cadl-ranch/issues/471

**Comment by qiaozha** [comment]
Can we move this out of this semester's scope ? it's not for green field GA.

**Comment by qiaozha** [comment]
@joheredi Should we close it? Since we have agreed to not support flatten in Modular?

**Comment by joheredi** [comment]
We discussed this, and we've decided to classify it as a "Won't Support" feature. It's not necessary for greenfield mpg, and we'll handle it as a known breaking change during brownfield library migrations, along with a few other breaking changes, including the LRO poller.
 
As a contingency plan, if we receive feedback that this breaking change is too disruptive during the brownfield migration, we can consider adding support as an additive change. However, our current plan is not to include it.
